src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,68 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX_RANDOM_NUMBER = 50;
+    private static final int FIGURES_COUNT = 5;
+    private static final String DEFAULT_COLOR = Color.WHITE.name();
+    private static final int DEFAULT_RADIUS = 10;
+    private Random random = new Random();
+
+    public Figure getRandomFigure() {
+        int randomIndex = random.nextInt(FIGURES_COUNT);
+        switch (randomIndex) {
+            case 0:
+                return getRandomCircle();
+            case 1:
+                return getRandomSquare();
+            case 2:
+                return getRandomRightTriangle();
+            case 3:
+                return getRandomRectangle();
+            case 4:
+                return getRandomIsoscelesTrapezoid();
+            default:
+                return getRandomIsoscelesTrapezoid();
+        }
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle(DEFAULT_COLOR, DEFAULT_RADIUS);
+    }
+
+    private Circle getRandomCircle() {
+        String randomColor = new ColorSupplier().getRandomColor();
```

Comment:
make a global ColorSupplier and use it in every method where you need it

Reply:
[colorSupplier to global](https://github.com/Maksym-Kulakov/jv-oop-advanced/commit/24d311f6cffb28137e1da82b30b1cd68285de836)